### PR TITLE
ci: add GitHub actions

### PR DESCRIPTION
**Pull request by ilanKushnir**

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,8450 +1,8 @@
 {
   "name": "prometheus-api-metrics",
   "version": "3.2.0",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
```
Comment (MatanAvneri) [inline package-lock.json]: Looks like the package-lock got changed to v1, can you revert this change?

Comment (ilanKushnir) [reply package-lock.json]: @MatanAvneri do you think its safe to change it manually?

Comment (MatanAvneri) [reply package-lock.json]: No, it should be only done by `npm` but I don't think it should be related to this MR anyway

Comment (ilanKushnir) [reply package-lock.json]: so do you want me to revert all package-lock changes?

Comment (ilanKushnir) [reply package-lock.json]: reverted, here is the workflow of this commit:
https://github.com/ilanKushnir/prometheus-api-metrics/actions/runs/895185399

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+          node-version: ${{ matrix.node }}
+      - name: install dependencies
+        run: npm ci
+      - name: install prom client
+        run: npm i prom-client@${{ matrix.prom-client }}
+      - name: run tests
+        run: npm test
+
+  release:
+    name: release
```
Comment (kobik) [inline .github/workflows/main.yml:95]: you need to grant the release job the required permissions as you can see here https://github.com/PayU/openapi-validator-middleware/blob/b37469d78236ba582de930af136074a7ba5f5950/.github/workflows/main.yml#L46

Comment (ilanKushnir) [reply .github/workflows/main.yml:95]: added.
@kobik I saw also some guy [adding '0.0.0-development'](https://youtu.be/rCXq86FOlzQ?t=567) to the package.json `version` in order to prevent people from touching it. should we do it too?

Comment (kobik) [reply .github/workflows/main.yml:95]: it's ok to set it to 0.0.0

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        run: npm run types-check
+      - name: lint check
+        run: npm run lint
+
+  test:
+    strategy:
+      matrix:
+        platform: [ ubuntu-latest ]
+        node: [ '10', '12', '14', '16' ]
+    needs: [ 'lockfile-lint', 'quality-checks' ]
```
Comment (kobik) [inline .github/workflows/main.yml]: the reason we require `lockfile-lint` before running the test is is to check that the lock file isn't compromised before running `npm install`.

i would actually add `lockfile-lint` to be needed by `quality-checks`

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        with:
+          path: '**/node_modules'
+          key: ${{ matrix.node }}-node-${{ hashFiles('**/package-lock.json') }}
+      - name: install dependencies
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: npm ci
+      - name: unit tests
+        run: npm run unit-tests
+      - name: integration tests
+        run: npx npm run integration-tests
```
Comment (kobik) [inline .github/workflows/main.yml]: there is a redundant `npx` command here

**Review by kobik — CHANGES_REQUESTED** [review]
thanks @ilanKushnir!

please see my comments

File: .github/workflows/main.yml (hunk 2 above)
Comment (ilanKushnir) [reply .github/workflows/main.yml:95]: set to `0.0.0`

File: .github/workflows/main.yml (hunk 3 above)
Comment (ilanKushnir) [reply .github/workflows/main.yml]: fixed

File: .github/workflows/main.yml (hunk 4 above)
Comment (ilanKushnir) [reply .github/workflows/main.yml]: removed

**Comment by ilanKushnir** [comment]
@kobik fixed, thanks!
pipeline: https://github.com/ilanKushnir/prometheus-api-metrics/actions/runs/913781142

**Review by kobik — APPROVED** [review]
